### PR TITLE
Add weekday pricing

### PR DESCRIPTION
**Pull request by MagnusHafstad**
### This pull request implements the requested changes to cabin pricing. 

- Prices can be different for weekdays and non week days
-

### CODE REVIEW

**Review by torgeirlysen — APPROVED** [review]
lgtm

File: frontend/src/components/pages/cabins/booking/Steps/calculatePrice.ts [hunk 1]
```diff
@@ -1,20 +1,44 @@
...
+      weekendNights++;
+    } else {
+      weekdayNights++;
+    }
+    currentDate = currentDate.add(1, "day");
   }
-  return NaN;
+  console.log("Weekday nights", weekdayNights);
+  console.log("Weekend nights", weekendNights);
+  console.log("Chosen cabins", chosenCabins);
```
Comment (larwaa) [inline frontend/src/components/pages/cabins/booking/Steps/calculatePrice.ts:33]: Doesn't matter all that much as we remove console.log for production regardless, but no need for these to stick around
```suggestion
```

File: frontend/src/components/pages/cabins/booking/Steps/calculatePrice.ts [hunk 2]
```diff
@@ -1,20 +1,44 @@
...
+  let totalPrice = 0;
+  while (currentDate.isBefore(finalDate, "day")) {
+    // Last day has no night.
+    if (currentDate.day() === 5 || currentDate.day() === 6) {
+      // 5 = Friday to sat, 6 = Saturday to sunday
+      weekendNights++;
+    } else {
+      weekdayNights++;
+    }
+    currentDate = currentDate.add(1, "day");
```
Comment (larwaa) [inline frontend/src/components/pages/cabins/booking/Steps/calculatePrice.ts:29]: I'm slightly concerned that we have to duplicate the backend logic here. I suggest that we instead add a `price`-query or something similar and rely on the backend logic to calculate the price. That way, we won't accidentally end up with different prices.

**Review by larwaa — COMMENTED** [review]
Admins can change prices at `frontend/src/pages/cabins/admin/index.tsx`, should also be updated so that they may change the prices for weekends. Minor comments otherwise

**Pull request merged**